packages/react/src/blocks/Video.tsx

```diff
@@ -37,25 +52,62 @@ class VideoComponent extends React.Component<{
 
   componentDidMount() {
     this.updateVideo();
+
+    if (this.lazyLoad && Builder.isBrowser) {
+      // TODO: have a way to consolidate all listeners into one timer
+      // to avoid excessive reflows
+      const listener = throttle(
+        (event: Event) => {
+          if (this.containerRef) {
+            const rect = this.containerRef.getBoundingClientRect();
```

Comment:
Do we support IE11? If not, we could use `IntersectionObserver` for this sort of logic: https://caniuse.com/#feat=intersectionobserver

Reply:
great idea

Reply:
I was hating the fact of how many timers we are creating that on fire trigger a reflow each so this would be a huge improvement